state-chain/cf-integration-tests/src/network.rs

```diff
@@ -643,8 +688,12 @@ impl Network {
 				.into_iter()
 				.map(|e| e.event)
 				.collect::<Vec<RuntimeEvent>>();
+
+			let cfe_events = state_chain_runtime::CfeEventEmitter::get_cfe_events(block_number)
+				.unwrap_or_default();
+
 			for engine in self.engines.values_mut() {
-				engine.handle_state_chain_events(&events);
+				engine.handle_state_chain_events(&events, &cfe_events);
```

Comment:
I think we should split these out, and have another processor for the runtime events, those not processed by the CFE. These integration tests have historically not been the easiest to follow, so anywhere we can make it a bit more logical, I think we should do it. Perhaps including this as a method on the network - which can then call it for each of the engines, for things like the key rotating.

This could wait for another PR though - as it may involve a bit of shifting around things in the tests.

Reply:
Sure, I'll save it for another PR.